keras/backend/mxnet_backend.py

```diff
@@ -2581,22 +2584,31 @@ def in_train_phase(x, alt, training=None):
         Either `x` or `alt` based on the `training` flag.
         the `training` flag defaults to `K.learning_phase()`.
     """
+    uses_learning_phase = False
+
     if training is None:
         training = learning_phase()
+        uses_learning_phase = True
 
-    if training is 1:
+    if training is 1 or training is True:
         if callable(x):
-            return x()
+            res = x()
         else:
-            return x
-    elif training is 0:
+            res = x
+        if isinstance(x, KerasSymbol):
+            uses_learning_phase = True
+    elif training is 0 or training is False:
         if callable(alt):
-            return alt()
+            res = alt()
         else:
-            return alt
+            res = alt
+    else:
```

Comment:
is this a real case? i think we only get here if training is int > 1

Reply:
training can be a Keras Placeholder.